### PR TITLE
Notification test

### PR DESCRIPTION
**Pull request by anshul2401**
Check if notification working or not from firestore.
-Go to grow/cloudMessaging.
- Tap on 'New Notification'
- Enter notification title and notification text.
-Click on next to go to Target section and choose your app.
- No changes in scheduling and conversion events.
- In additional option, provide key value pair as follow
   - 'click_action': 'FLUTTER_NOTIFICATION_CLICK'
  - 'id': '1'
  - 'from':'A'
  - 'to':'B'
  - 'containerName':'ABC'
  - 'pickupTime': '9AM'
- click on review and then publish
new notification should be generated

### CODE REVIEW

**Comment by anshul2401** [comment]
Check if notification is being generated or not then i will procced further.

**Comment by harsimranmaan** [comment]
Thanks, Will start integrating this soon

File: android/app/src/main/kotlin/ca/bigpanther/shipanther/Application.kt [hunk 1]
```diff
@@ -0,0 +1,18 @@
+package com.example.notificationTest
```
Comment (harsimranmaan) [inline android/app/src/main/kotlin/ca/bigpanther/shipanther/Application.kt:1]:
```suggestion
package com.example.notificationTest
package ca.bigpanther.shipanther
```

File: android/app/src/main/kotlin/ca/bigpanther/shipanther/Application.kt [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+import io.flutter.app.FlutterApplication
+import io.flutter.plugin.common.PluginRegistry
+import io.flutter.plugin.common.PluginRegistry.PluginRegistrantCallback
+import io.flutter.plugins.GeneratedPluginRegistrant
+import io.flutter.plugins.firebasemessaging.FlutterFirebaseMessagingService
+import io.flutter.plugins.firebasemessaging.FirebaseMessagingPlugin
+class Application:FlutterApplication(), PluginRegistrantCallback {
```
Comment (harsimranmaan) [inline android/app/src/main/kotlin/ca/bigpanther/shipanther/Application.kt:8]: AndroidManifest.xml should be modified with `<application android:name=".Application"`

File: lib/screens/detail_screen.dart [hunk 3]
```diff
@@ -41,6 +47,90 @@ class DetailScreenState extends State<DetailScreen> {
     super.dispose();
   }
 
+  //Notifications starts here
+
+  FirebaseMessaging _firebaseMessaging = FirebaseMessaging();
```
Comment (harsimranmaan) [inline lib/screens/detail_screen.dart:52]: I think the messaging instance should be initialized in main.dart

File: android/app/src/main/kotlin/ca/bigpanther/shipanther/Application.kt (hunk 2 above)
Comment (harsimranmaan) [reply android/app/src/main/kotlin/ca/bigpanther/shipanther/Application.kt:8]: Added this in the new commits

File: lib/screens/detail_screen.dart (hunk 3 above)
Comment (harsimranmaan) [reply lib/screens/detail_screen.dart]: Added this in the new commits

**Comment by harsimranmaan** [comment]
Tested notifications when the app is opened or in background

notification

```
I/flutter ( 8515): onMessage: {notification: {title: Test, body: test6}, data: {pickupTime: 9AM, id: 1, to: B, click_action: FLUTTER_NOTIFICATION_CLICK, 'from': A, containerName: ABC}}
W/FirebaseMessaging( 8515): Missing Default Notification Channel metadata in AndroidManifest. Default value will be used.
I/flutter ( 8515): onResume: {notification: {}, data: {google.original_priority: high, google.sent_time: 1605594091057, google.delivered_priority: high, 'from': A, click_action: FLUTTER_NOTIFICATION_CLICK, google.message_id: 0:1605594091351598%f3dc942af3dc942a, collapse_key: ca.bigpanther.shipanther, pickupTime: 9AM, containerName: ABC, google.ttl: 2419200, from: 219386321464, id: 1, to: B}}
```

**Review by harsimranmaan — APPROVED** [review]
verified with changes

**Pull request merged**